judge/views/contests.py

```diff
@@ -93,13 +94,17 @@ def get_context_data(self, **kwargs):
                     active.append(participation)
                     present.remove(participation.contest)
 
+        start = self.paginate_by * (context['page_obj'].number - 1)
+        past = past[start:start+self.paginate_by]
```

Comment:
We definitely cannot do this. Suggestion is to paginate the queryset for past contests.